app/models/package_manager/maven.rb

```diff
@@ -83,13 +83,26 @@ def self.mapping_from_pom_xml(version_xml)
       else
         xml = version_xml
       end
-      {
-        description: xml.locate('description').first.try(:nodes).try(:first),
-        homepage: xml.locate('url').first.try(:nodes).try(:first),
-        repository_url: repo_fallback(xml.locate('scm/url').first.try(:nodes).try(:first),
-                                      xml.locate('url').first.try(:nodes).try(:first)),
+
+      keys = %i[description homepage repository_url licenses]
+      parent = Hash[keys.product([nil])]
+      if xml.locate('parent').first
+        group_id = xml.locate('parent/groupId').first&.nodes&.first
+        artifact_id = xml.locate('parent/artifactId').first&.nodes&.first
+        if group_id && artifact_id
+          parent = mapping(project([group_id, artifact_id].join(":")))
```

Comment:
I wonder if we want to have a maxdepth to avoid someone doing an incredibly deep pom nesting and causing us probelms

Reply:
I don't think that would be an issue, but it is something to protect against. What would a reasonable depth limit be? I've never seen a package with more than 2-3 parent projects. Maybe setting the max depth to something like 5?

Reply:
5 sounds totally good to me